File: backend/FluffitMember/src/main/java/com/ssafy/fluffitmember/messagequeue/KafkaProducer.java
package com.ssafy.fluffitmember.messagequeue;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.ssafy.fluffitmember.exercise.dto.StepsKafkaDto;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class KafkaProducer {

    @Autowired
    private KafkaTemplate<String, String> kafkaTemplate;

    public StepsKafkaDto send(String topic, StepsKafkaDto stepsKafkaDto) {
        ObjectMapper mapper = new ObjectMapper();
        String jsonInString = "";
        try {
            jsonInString = mapper.writeValueAsString(stepsKafkaDto);
        } catch(JsonProcessingException ex) {
            ex.printStackTrace();
        }

        kafkaTemplate.send(topic, jsonInString);
        log.info("Kafka Producer sent data from the Order microservice: " + stepsKafkaDto);

        return stepsKafkaDto;
    }
}

File: backend/FluffitFlupet/src/main/kotlin/com/ssafy/fluffitflupet/messagequeue/KafkaProducer.kt
package com.ssafy.fluffitflupet.messagequeue

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import com.ssafy.fluffitflupet.dto.CoinKafkaDto
import com.ssafy.fluffitflupet.service.FlupetService
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class KafkaProducer(
    private val kafkaTemplate: KafkaTemplate<String, String>
) {
    private val log = LoggerFactory.getLogger(FlupetService::class.java)

    fun send(topic: String, coinKafkaDto: CoinKafkaDto) {
        val mapper = ObjectMapper()
        val jsonInString: String = try {
            mapper.writeValueAsString(coinKafkaDto) //객체를 json 포맷으로 변환
        }catch (ex: JsonProcessingException){
            ex.printStackTrace()
            return
        }
        kafkaTemplate.send(topic, jsonInString)
        log.info("Kafka Producer sent data from the Order microservice: $coinKafkaDto")
    }
}